src/librustc_mir/transform/check_consts/validation.rs

```diff
@@ -588,7 +588,13 @@ impl Visitor<'tcx> for Validator<'mir, 'tcx> {
                 };
 
                 if needs_drop {
-                    self.check_op_spanned(ops::LiveDrop, err_span);
+                    let term_span = terminator.source_info.span;
+                    let op = if term_span.hi() - term_span.lo() > rustc_span::BytePos(1) {
```

Comment:
I think what you want is `!term_span.is_dummy()` at which point you should be able to not do this branch at all, as the diagnostics system can handle dummy spans I think

Reply:
The ui test diffs seem good to me and you can probably just bless them.

Can you also add a test showing your example in the PR post?

Reply:
I was just following @ecstatic-morse suggestion: https://github.com/rust-lang/rust/issues/72907#issuecomment-637821922

Also without it there are a lot of overlapping messages, let me get you an example